connect/resources/automation_engine.py

```diff
@@ -16,19 +15,23 @@
 
 class AutomationEngine(BaseResource):
     limit = 1000  # type: int
-    logger = logging.getLogger()
+
+    def __init__(self, config=None):
+        super(AutomationEngine, self).__init__(config)
+        self._current_request = None
+        self._logger_adapter = None
 
     def filters(self, status='pending', **kwargs):
         # type: (str, Dict[str, Any]) -> Dict[str, Any]
         return super(AutomationEngine, self).filters(status=status, **kwargs)
 
-    @function_log(custom_logger=logger)
+    @function_log
     def process(self, filters=None):
-        '''
-        # type: (Dict[str, Any]) -> None
-        '''
+        # # type: (Dict[str, Any]) -> None
         for request in self.list(filters):
+            self._set_current_request(request)
```

Comment:
For each request, the request to process is stored so the correct prefix is used in the LoggerAdapter of the AutomationEngine instance. This approach works fine with multithreading.